NRLMMD-GEOIPS/geoips
# v1.9.0 geoips - VERSION RELEASE - versioning, test repo, documentation updates

# Related Issues
required to close NRLMMD-GEOIPS/geoips#105

# Reviewer Instructions
* Please confirm v1.8.0.dev1 changes are included in v1_8_0_dev1.rst release notes
   * https://github.com/NRLMMD-GEOIPS/geoips/releases/tag/1.8.0.dev1
   * https://github.com/NRLMMD-GEOIPS/geoips/blob/v1.9.0-release/docs/source/releases/v1_8_0_dev1.rst
* Confirm the last closed PR is included in the 1.8.0.dev1 release notes
   * https://github.com/NRLMMD-GEOIPS/geoips/pulls?q=is%3Apr+is%3Aclosed
* Review "Files changed" tab
* Confirm appropriate testing was completed for these updates.

# Summary
Changes for version 1.9.0 update on repo geoips.

## What's Changed v 1.9.0
### Breaking Changes 🛠
* v1.8.0 geoips - OPEN SOURCE INCOMING - plugin interface updates
* Replace SSMI/S 89GHz product names with 91GHz
### Exciting New Features 🎉
* Add productWebPath to TC products and unsectored schema
* Add geoips_clavrx to test_full_install.sh
### Regression Fixes
* Correct minor sphinx build warnings 
### Bug Fixes
* Resolve F-16 SSMI/S 37GHz image issue
* Resolve basename: extra operands error in tests
* Bug fix: Update SSMI/S color89 script in test_all to color91
* Fix permissions on setup.sh and test_full_install.sh in open-source-delivery branch 
### Installation Updates
* Add prettier and doc8 to setup.py for YAML and RST linting
* Use setuptools_scm for versioning
* Use pip installed matplotlib >= 3.7.0
* Update test repo outputs for latest dependency versions
### Code Formatting / Style / Refactoring Updates
* Move database writes to geoips_db
### Git Workflow Improvements
* Add "test repo udpates" label
### Testing Updates
* Add test calls to code_checks.sh and build_docs.sh
* Update test full install.sh to install geoips plugins in addition to cloning repos
### Documentation Updates
* Have documentation build return non zero when not successful

Changes for v1.7.0...

## What's Changed v 1.7.0
### Security Updates
* Security Updates based on medium and high bandit security issues 
### Exciting New Features 🎉
* Add resource usage statistics table
* Add data start, end, and mid datetime to TC YAML metadata output
### Regression Fixes
* Update SAR reader for new coverage start and end time format
### Bug Fixes
* Replace xarray_obj.order with fuse_order
* Fix minor issues that occur while building documentation
### Installation Updates
* Move requirements from extras require to install requires
### Git Workflow Improvements
* Update release.yml for generating Release Notes
### Testing Updates
* Updates to syntax checking, minor documentation updates
* xrdict_product_area_outlist to product.py, data_times to AMSR2 YAML metadata, Error and Number expected files to test log output
* Extend SAR incident angle plots to 70 degrees
### Documentation Updates
* v1.6.1 geoips - OPEN SOURCE DELIVERY internal updates - relative doc links
* 100 doc sphinx nrl template
* Sphinx automodule API documentation build
* Remove "2.0" from documentation
* RST Outline of full documentation organization
* Pass full package path when building documentation, and allow html-only build.
* Prepare updated documentation for open source release
* Update individual release notes from md to rst format.
* Update CHANGELOG.md to CHANGELOG.rst 
### Other Changes
* v1.6.3 geoips - INTERNAL RELEASE - resource stats, docs, code checks 
* v1.7.0 geoips - OPEN SOURCE DELIVERY 


See NRLMMD-GEOIPS/geoips#105 for additional repositories included in this update.

# Testing Instructions

Install GEOIPS and clone v1.9.0-release branch for:
* geoips


Tests should return 0:
$GEOIPS_PACKAGES_DIR/geoips/tests/test_full_install.sh


# Post Merge Steps
Once this PR has been merged, v1.9.0 will be tagged/released on the geoips main branch.
